ML-model/Task1-Data_scraping/web_scraper.py

```
@@ -1,30 +1,54 @@
+
+    # Open the webpage
+    driver.get(url)
+
+    try:
+        # Wait for the results to load (adjust the wait condition based on the content you're waiting for)
+        WebDriverWait(driver, 10).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR, 'div.List-results-item'))
+        )
+        
+        # Once loaded, get the page source and parse it with BeautifulSoup
+        soup = BeautifulSoup(driver.page_source, 'html.parser')
```

Ruff (0.8.2)

Undefined name `BeautifulSoup`

(F821)